traffic_ops/app/lib/API/User.pm

```diff
@@ -376,6 +387,10 @@ sub get_deliveryservices_not_assigned_to_user {
 
 	my $rs_links = $self->db->resultset("Deliveryservice")->search( undef, { order_by => "xml_id" } );
 	while ( my $row = $rs_links->next ) {
+		if (!$tenant_utils->is_ds_resource_accessible_to_tenant($tenants_data, $row->tenant_id, $user->tenant_id)) {
```

Comment:
@mitchell852 
We do not check the tenancy of the current user here.
Do you want to add it?